app/src/test/java/org/simple/clinic/settings/changelanguage/ChangeLanguageUpdateTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.simple.clinic.settings.changelanguage
+
+import com.spotify.mobius.test.NextMatchers.hasEffects
+import com.spotify.mobius.test.NextMatchers.hasModel
+import com.spotify.mobius.test.NextMatchers.hasNoEffects
+import com.spotify.mobius.test.NextMatchers.hasNoModel
+import com.spotify.mobius.test.UpdateSpec
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.settings.Language
+
+@RunWith(JUnitParamsRunner::class)
+class ChangeLanguageUpdateTest {
+
+  private val defaultModel = ChangeLanguageModel.FETCHING_LANGUAGES
+
+  private val englishIndia = Language(displayName = "English", languageCode = "en_IN")
+  private val hindiIndia = Language(displayName = "हिंदी", languageCode = "hi_IN")
+
+  private val spec = UpdateSpec<ChangeLanguageModel, ChangeLanguageEvent, ChangeLanguageEffect>(ChangeLanguageLogic::update)
+
+  @Test
+  @Parameters(method = "params for initial models when loading selected language")
+  fun `when the selected language is loaded, the model must be updated`(model: ChangeLanguageModel) {
+    spec
+        .given(model)
+        .whenEvent(CurrentSelectedLanguageLoadedEvent(englishIndia))
+        .then(assertThatNext(
+            hasModel(model.withCurrentLanguage(englishIndia)),
+            hasNoEffects()
+        ))
+  }
+
+  @Suppress("Unused")
+  private fun `params for initial models when loading selected language`(): List<ChangeLanguageModel> {
+    return listOf(
+        defaultModel,
+        defaultModel.withSupportedLanguages(listOf(englishIndia, hindiIndia))
```

Comment:
When we include parameters, we are coupling state transitions to the test itself. This is a lot of cognitive overhead to process (in this case the maintainer should also remember the `init` tests). It it better to break this test down make them explicit. Also, I would avoid parameterized tests in `init` and `update` tests altogether. This could even be a lint rule.

Reply:
See earlier comment.